enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,347 @@
+
+1. The cluster creator provides an install-config and credentials
+2. (optional) The cluster creator runs `openshift-install create manifests`
+3. (optional) The cluster creator edits the newly created CAPI manifests.
+4. The cluster creator runs `openshift-install create cluster`
+5. `openshift-install` extracts binaries for kube-apiserver, etcd, CAPI infrastructure provider & cloud CAPI provider to the install dir
+6. `openshift-install` using `envtest` initializes a control plane locally on the Installer host
+7. `openshift-install` execs the CAPI infrastructure and cloud provider as subprocesses, pointing them to the local control plane
+8. `openshift-install` applies the CAPI manifests to the control plane
+9. The CAPI controllers provision cluster infrastructure based on the manifests
+10. `openshift-install` monitors the status of the local manifests as they are applied
+11. If the statuses are as expected, infrastrucutre has been provisioned and installation continues with the normal flow.
+
+In the case of an error in the final step, the Installer will bubble up resources with non-expected statuses.
+
+#### Variation and form factor considerations [optional]
+
+
+### API Extensions
+
+As a result of this enhancement, API Extensions will only be used locally by the 
+local Installer control plane--they will not, until future work, be pivoted to the cluster.
+The following are CRD references for Installer-supported platforms:
+
+- AWS: [infrastructure.cluster.x-k8s.io/v1beta2][awsCRD]
+- Azure/AzureStack: [infrastructure.cluster.x-k8s.io/v1beta1][azureCRD]
+- GCP: [infrastructure.cluster.x-k8s.io/v1beta1][gcpCRD]
+- IBMCloud/PowerVS: [infrastructure.cluster.x-k8s.io/v1beta2][ibmCRD]
+- Nutanix: [infrastructure.cluster.x-k8s.io/v1beta1][nutanixCRD]
+- OpenStack: [infrastructure.cluster.x-k8s.io/v1alpha7][openstackCRD]
+- vSphere: [infrastructure.cluster.x-k8s.io/v1beta1][vsphereCRD]
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Overview
+
+In a typical CAPI installation, manifests indicating the desired cluster configuration are applied to a
+management cluster. In order to keep `openshift-install` free of new external runtime dependencies,
+the management-cluster dependencies will be [embedded][embed] into the `openshift-install` binary,
+extracted at runtime, and cleaned up afterward.  This approach is similar to what we have been using
+for Terraform.
+
+In its current configuration, the Installer embds the Terraform and cloud-specific provider binaries
+within the `openshift-install` binary and extracts them at runtime. The Installer produces the Terraform
+configuration files and invokes Terraform using the `tf-exec` library.
+
+![alt terraform diagram](terraform_embedded.jpg)
+
+
+We can follow a similar pattern to run CAPI controllers locally on the Installer host. In addition
+to the CAPI controller binaries, `kube-apiserver` and `etcd` are embedded in order to run a local
```

Comment:
what that running on the boostrap or on each node?

Reply:
They'll run on whatever system is running the `openshift-install` binary. So perhaps the laptop of the person who is installing openshift.